### PR TITLE
meta-mbl-distro: Introduce debug interface setting for production images

### PR DESCRIPTION
**Pull request by diego-sueiro**
We introduced variable MBL_PRODUCTION_ETH_DBG set in
meta-mbl-distro/conf/distro/include/mbl-distro-production.inc file which
controls whether or not we want the production images to have an
(usbgadget) ethernet debug interface.
The (usbgadget) ethernet debug interface name is set in the
${MACHINE}.conf file via MBL_DBG_IFNAME variable.

The 'production-dbg' PACKAGECONFIG option was introduced to change
the behavior of the following recipes:

- packagegroup-mbl-production: To install systemd-net-conf-dbg-iface
  and dropbear.

- dropbear: To install a dropbear.socket file that only allows ssh on
  the MBL_DBG_IFNAME interface and enable root login.

- systemd-net-conf: To install systemd network files to restrict mDNS
  to the MBL_DBG_IFNAME interface.

As a part of this patch, the systemd-net-conf-dbg-iface recipe was
updated to include the systemd network link file to rename the
usbgadget ethernet interface to MBL_DBG_IFNAME and avoid naming
conflicts to other network devices that get the usb[n] pattern name.

### CODE REVIEW

File: meta-mbl-distro/recipes-core/systemd/systemd-net-conf.bb [hunk 1]
```diff
@@ -10,11 +10,19 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
...
+           ${@bb.utils.contains('PACKAGECONFIG','development-dbg','file://10-mbl-development.network','',d)} \
+           ${@bb.utils.contains('PACKAGECONFIG','production-dbg','file://99-mbl-production.network','',d)} \
+           ${@bb.utils.contains('PACKAGECONFIG','production-dbg','file://10-mbl-production-dbg.network','',d)} \
+           ${@bb.utils.contains('PACKAGECONFIG','production-dbg','file://llmnr.conf','',d)} \
 "
 
+PACKAGECONFIG ??= "development-dbg"
+
+PACKAGECONFIG[development-dbg] = ""
+PACKAGECONFIG[production-dbg] = ""
```
Comment (jonathanhaigh-arm) [inline meta-mbl-distro/recipes-core/systemd/systemd-net-conf.bb]: Discussed with @diego-sueiro in person. I don't think `development-dbg` and `production-dbg` are the best names for these config options. I think the names should describe what the option actually does.

Comment (jonathanhaigh-arm) [reply meta-mbl-distro/recipes-core/systemd/systemd-net-conf.bb]: ...Discussed with @diego-sueiro in person some more. Splitting up the different aspects of the network config into descriptive `PACKAGECONFIG` items would be a bit messy. And as Diego pointed out, this recipe is a config recipe itself so it's not too bad if it contains production v.s. development policy decisions.

**Comment by diego-sueiro** [comment]
New jenkins build: http://jenkins.mbed-linux.arm.com/job/ds-warrior-dev/97/

File: meta-mbl-distro/recipes-core/systemd/systemd-net-conf.bb (hunk 1 above)
Comment (diego-sueiro) [reply meta-mbl-distro/recipes-core/systemd/systemd-net-conf.bb]: Changed in d8ccf66.

File: meta-mbl-distro/recipes-core/packagegroups/packagegroup-mbl-production.bb [hunk 2]
```diff
@@ -57,4 +57,10 @@ PACKAGEGROUP_MBL_PRODUCTION_PKGS_append = " connman"
 PACKAGEGROUP_MBL_PRODUCTION_PKGS_append = " connman-client"
 PACKAGEGROUP_MBL_PRODUCTION_PKGS_append = " systemd-net-conf"
 
+# Packages to be installed when 'production-dbg' PACKAGECONFIG is set
+PACKAGEGROUP_MBL_PRODUCTION_PKGS_append = " \
+    ${@ bb.utils.contains('COMBINED_FEATURES', 'usbgadget', 'systemd-net-conf-dbg-iface', '', d) if bb.utils.contains('PACKAGECONFIG', 'production-dbg', 'True', '', d) else ''} \
+    ${@ bb.utils.contains('PACKAGECONFIG', 'production-dbg', 'dropbear', '', d)} \
```
[the remark this answers is not in the document]
Comment (diego-sueiro) [reply meta-mbl-distro/recipes-core/packagegroups/packagegroup-mbl-production.bb]: Changed in d8ccf66.

File: meta-mbl-distro/recipes-core/dropbear/dropbear_%.bbappend [hunk 3]
```diff
@@ -4,7 +4,10 @@
 
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
+PACKAGECONFIG[production-dbg] = ""
```
[the remark this answers is not in the document]
Comment (diego-sueiro) [reply meta-mbl-distro/recipes-core/dropbear/dropbear_%.bbappend]: Changed in d8ccf66.

**Pull request merged**